src/Delegation.sol

```diff
@@ -358,6 +384,44 @@ contract Delegation is IDelegation, EIP712, GuardedExecutor {
         return true;
     }
 
+    /// @dev Can be used to pause/unpause the contract, in case of emergencies.
+    /// - Pausing the contract will make all signature validations fail,
+    ///   effectively blocking all pay, execute, isValidSignature attempts.
+    /// - The `pauseAuthority` can unpause the contract at any time.
+    /// - Anyone can unpause the contract after the PAUSE_EXPIRY has passed.
+    /// - Note: Contracts CANNOT be paused again until PAUSE_EXPIRY + 1 week has passed.
+    ///   This is done to prevent griefing attacks, where a malicious pauseAuthority,
+    ///   keeps censoring the user.
+    function pause(bool isPause) public virtual {
+        DelegationStorage storage $ = _getDelegationStorage();
+        if (isPause) {
+            // Owners can use this 1 week buffer, to update their `pauseAuthority` if needed.
+            if (
+                msg.sender != $.pauseAuthority
+                    || block.timestamp < $.lastPauseTimestamp + PAUSE_EXPIRY + 1 weeks
+                    || $.flags.get(PAUSE_FLAG)
+            ) {
+                revert Unauthorized();
+            }
```

Comment:
Imo, `$.lastPauseTimestamp + PAUSE_EXPIRY` into a temp variable `pauseUntil`. Then `forge fmt` will hopefully make this into a single line.